piptools/scripts/compile.py

```diff
@@ -143,6 +143,8 @@ def cli(verbose, dry_run, pre, rebuild, find_links, index_url, extra_index_url,
         repository = LocalRequirementsRepository(existing_pins, repository)
 
     log.debug('Using indexes:')
+    # remove duplicate index urls before processing
+    repository.finder.index_urls = sorted(set(repository.finder.index_urls))
```

Comment:
Thanks for your contribution!

As it is, this has the possibility of changing the index-url order, which is a property that needs to be maintained.

As a user, I might use index urls to explicitly set desired index usage ( A -> B -> C ).

See http://stackoverflow.com/a/39835527/28275 for a clean reference implementation that keeps order.